model/riscv_insts_vext_total.sail

```diff
@@ -0,0 +1,9655 @@
+    /* Least significant SEW bits */
+    slice(X(rs1), 0, vsew_bits)
+  }
+  else if sizeof(xlen) < vsew_bits then {
+    /* Sign extend to SEW */
+    sail_sign_extend(X(rs1), vsew_bits)
+  }
+  else {
+    X(rs1)
+  }
+}
+
+/* Get the starting element index from csr vtype */
+val get_start_element : unit -> int effect {escape, rreg, wreg}
+function get_start_element() = {
+  start_element     : int = unsigned(vstart);
+  vsew_bits : int = get_vtype_vsew();
+  //print("start_element0=" ^ BitStr(to_bits(sizeof(xlen),start_element)));
+  //print("element_width0=" ^ BitStr(to_bits(sizeof(xlen),vsew_bits)));
+  /* The use of vstart values greater than the largest element
+       index for the current SEW setting is reserved.
+     It is recommended that implementations trap if vstart is out of bounds.
+       It is not required to trap, as a possible future use of upper vstart bits
+       is to store imprecise trap information. */
+  if start_element > ((8 * sizeof(vlen) / vsew_bits) - 1) then {
+    //handle_illegal()
+    start_element = -1;
+    print("illegal start_element or luml")
+  };
+  //print("end get");
+  start_element
+}
+
+/* Get the ending element index from csr vl */
+val get_end_element : unit -> int effect {escape, rreg, wreg}
+function get_end_element() = {
+  let end_element : int = unsigned(vl) - 1;
+  end_element
+}
+
+/* Mask handling; creates a pre-masked result vector for vstart, vl, vta/vma, and vm */
+/* vm should be baked into vm_val from doing read_vmask */
+/* tail masking when lmul < 1 is handled in write_vreg */
+/* Returns two vectors:
+ *   vector1 is the result vector with values applied to masked elements
+ *   vector2 is a "mask" vector that is true for an element if the corresponding element
+ *     in the result vector should be updated by the calling instruction
+ */
+val init_masked_result : forall 'n 'm, 8 <= 'm <= 128. (int('n), int('m), real, vector('n, dec, bits('m)), vector('n, dec, bool)) -> (vector('n, dec, bits('m)), vector('n, dec, bool)) effect {escape, rreg, undef, wreg}
+function init_masked_result(num_elem, vsew_bits, lmul, vd_val, vm_val) = {
+  let start_element : int                       = get_start_element();
+  let end_element   : int                       = get_end_element();
+  let tail_ag       : agtype                    = get_vtype_vta();
+  let mask_ag       : agtype                    = get_vtype_vma();
+  mask_helper       : vector('n, dec, bool)     = undefined;
+  result            : vector('n, dec, bits('m)) = undefined;
+  //print_int("start_element",start_element);
+  //print_int("end_element",end_element);
+
+  if start_element < 0 then {
+    /* start element is not valid */
+    result      = undefined;
+    mask_helper = undefined;
+  }
+  else {
+    /* Determine the actual number of elements when lmul < 1 */
+    var real_num_elem : int = undefined;
+    if lmul >= 1.0 then {
+      real_num_elem = num_elem;
+      //print("init_masked_result: lmul >= 1.0")
```

Comment:
Debug info like this need to be removed before merging ? 

If there exist any debug information that we think would be very useful at a later stage, then we perhaps should have them guarded by a higher verbosity debug print? 

I don't think the model has variable verbosity debug logging at the moment, but perhaps that's something someone could argue that is needed  Debug info like this need to be removed before merging ? 

If there exist any debug information that we think would be useful at a later stage, then we perhaps should have them guarded by a higher verbosity debug print? 

I don't think the model has variable verbosity debug logging at the moment, but perhaps that's something someone could argue that is needed  @Debug info like this need to be removed before merging ? 

If there exist any debug information that we think would be useful at a later stage, then we perhaps should have them guarded by a higher verbosity debug print? 

I don't think the model has variable verbosity debug logging at the moment, but perhaps that's something someone could argue that is needed  @billmcspadden-riscv

Reply:
Yes,  these kind of "debug droppings" should be removed or,  if they are potentially useful,  then a verbosity level should be added.

The SystemVerilog UVM package has a pattern we could follow.   Is that something that people would respect and use?